frontend/public/components/catalog.jsx

```diff
@@ -16,6 +16,10 @@ class CatalogListPage extends React.Component {
 
     this.state = {
       items: [],
+      counts: {
+        clusterServiceClasses: 0,
+        imageStreams: 0
```

Comment:
It's too bad you have to pass this in, although I do think this is still cleaner than before.

An alternative is to count items by type in `CatalogTileViewPage` when items change to keep the interface cleaner. But I'm OK with this.

Reply:
yeah, I thought of that as well, but the counts are available in `catalog.jsx`.  If I do it in `CatalogTileViewPage`, I'd have to `_.filter` by `kind` and take more processing time to calc the counts again.

Reply:
It doesn't change the overall time complexity since we're already sorting when items change, and it would be a cleaner abstraction.

You should be able to use something like `_.countBy(items, 'obj.kind')` to get counts.

https://lodash.com/docs/4.17.10#countBy